.github/workflows/check-merge-labels.yml

```diff
@@ -0,0 +1,33 @@
+name: Check merge labels
+
+on:
+  pull_request:
+    types: [labeled, unlabeled]
+  merge_group:
+permissions: read-all
+
+jobs:
+  label_checks:
+    runs-on: ubuntu-latest
+    steps:
+      - id: explicitly_disallowed
+        if: |
+          contains(github.event.pull_request.labels.*.name, 'do not merge')
+        run: |
+          echo "This is labeled \"Do not merge\"."
+          exit 1
+
+      - id: future_release
+        if: |
+          contains(github.event.pull_request.labels.*.name, 'release-0.9')
+        run: |
+          echo "This is targeted for a future release."
+          exit 1
+
+      - id: breaking_change
+        if: |
+          contains(github.event.pull_request.labels.*.name, 'abi-break') ||
+          contains(github.event.pull_request.labels.*.name, 'breaking change')
+        run: |
+          echo "This is a breaking change."
+          exit 1
```

Comment:
I made these all `steps` instead of e.g. different `jobs` because it cluttered up the list of checks in the PR view, but I don't feel strongly about it if there's a better organization.